packages/react/src/Icon/Icon.tsx

```diff
@@ -7,15 +7,15 @@
 
 import clsx from 'clsx'
 import { forwardRef } from 'react'
-import type { ForwardedRef, HTMLAttributes } from 'react'
+import type { ForwardedRef, HTMLAttributes, ReactNode } from 'react'
 
 export type IconProps = {
   /** The size of the icon. Corresponds with the text levels. */
   size?: 'level-3' | 'level-4' | 'level-5' | 'level-6'
   /** Whether the icon container should be made square. */
   square?: boolean
   /** The component rendering the icon’s markup. */
-  svg: Function
+  svg: Function | ReactNode
```

Comment:
Now you can pass any component to Icon, you also could've used HeadMenuIcon.svg or prevented Menu.svg from combining paths.

Reply:
I think this change is useful for other uses of the Icon component. Passing a function still works, but if you need to pass a React component with props / a classname, you can.